app/models/concerns/legal_aid_application_state_machine.rb

```diff
@@ -50,7 +50,10 @@ module LegalAidApplicationStateMachine
 
       event :complete_means do
         transitions from: :checking_citizen_answers, to: :means_completed,
-                    after: -> { CleanupCapitalAttributes.call(self) }
+                    after: -> do
+                      CleanupCapitalAttributes.call(self)
+                      update!(provider_step: :client_received_legal_helps)
```

Comment:
The should be sufficient for the feature to work. Do you still need the changes to the other files?

Reply:
Done thanks